clusterloader2/api/validation.go

```diff
@@ -105,6 +108,35 @@ func (v *ConfigValidator) validateStep(s *Step, fldPath *field.Path) field.Error
 	return allErrs
 }
 
+func (v *ConfigValidator) handleTuningSetInPhase() field.ErrorList {
+	allErrs := field.ErrorList{}
+	c := v.config
+	for i := range c.Steps {
+		allErrs = append(allErrs, validateTuningSetInPhase(c.Steps[i], c.TuningSets, field.NewPath("step").Index(i))...)
+	}
+	return allErrs
+}
+
+func validateTuningSetInPhase(s *Step, ts []*TuningSet, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	for i := range s.Phases {
+		if s.Phases[i].TuningSet != "" {
+			found := false
+			for j := range ts {
+				if s.Phases[i].TuningSet == ts[j].Name {
+					found = true
+					break
+				}
+			}
+			if found == false {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("phase").Index(i).Child("tuningSet"), s.Phases[i].TuningSet, "tuning set referenced has not been declared"))
```

Comment:
Sorry, I've missed it. It should be "phases" instead of "phase"

Reply:
Done!